File: Práctica 2/Queue.java
public class Queue {
    private List data;

    public Queue() {
        data = new List();
    }

    public int size() {
        return data.getSize();
    }

    public boolean isEmpty() {
        return data.isEmpty();
    }

    public void enqueue(Object e) {
        data.addLast(e);
    }

    public Object dequeue() {
        if (isEmpty()) {
            return null;
        }
        return data.removeFirst();
    }

    public Object first() {
        if (isEmpty()) {
            return null;
        }
        return data.First().getData();
    }
}

File: Práctica 3/Estructuras/Queue.py
from Estructuras.List import List


class Queue:
    def __init__(self):
        self.data = List()

    def size(self):
        return self.data.getSize()

    def isEmpty(self):
        return self.data.isEmpty()

    def enqueue(self, e):
        self.data.addLast(e)

    def dequeue(self):
        if self.isEmpty():
            return None
        return self.data.removeFirst()

    def first(self):
        if self.isEmpty():
            return None
        return self.data.First().getData()
